kaldb/src/main/java/com/slack/kaldb/chunkManager/DiskOrMessageCountBasedRolloverStrategy.java

```diff
@@ -0,0 +1,101 @@
+package com.slack.kaldb.chunkManager;
+
+import static com.slack.kaldb.util.ArgValidationUtils.ensureTrue;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.slack.kaldb.proto.config.KaldbConfigs;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.io.File;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a disk based rollover strategy that calculates disk size every 10 seconds The disk size
+ * being calculated in the lucene index at that point in time. In addition, if we hit the max messages limit we also rollover
+ */
+public class DiskOrMessageCountBasedRolloverStrategy implements ChunkRollOverStrategy {
```

Comment:
would have liked this to be disk based only.

However since this is the default strategy, tests like `KaldbTest` which want to add exactly N message and perform a rollover is tough to fix with just a disk based rollover strategy

In practice, we'll keep the maxMessagesPerChunk high like 2B or something